scripts/sudo.ps1

```diff
@@ -1,87 +1,78 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
+[CmdletBinding(DefaultParameterSetName = "Script")]
+param(
+	[Parameter(Mandatory, Position = 0, ParameterSetName = "Script")]
+	[scriptblock]$ScriptBlock,
 
-# open question - Should -NoProfile be used when invoking PowerShell
-BEGIN {
+	[switch]$NoProfile,
+
+	[Parameter(Mandatory, Position = 0, ParameterSetName = "Command")]
+	[string]$Command,
+
+	[Parameter(Position = 1, ValueFromRemainingArguments)]
+	[Alias("Args")]
+	[PSObject[]]$ArgumentList
+)
+begin {
 	if ($__SUDO_TEST -ne $true) {
-		$SUDOEXE = "sudo.exe"
-	}
-	else {
-		if ($null -eq $SUDOEXE) {
-			throw "variable SUDOEXE has not been set for testing"
-		}
+		$Env:SUDOEXE = "sudo.exe"
+	} elseif (!$Env:SUDOEXE) {
+			throw "Environment variable SUDOEXE has not been set for testing"
 	}
 
-	if ([Environment]::OSVersion.Platform -ne "Win32NT") {
+	if ($IsLinux -or $IsMacOS) {
 		throw "This script works only on Microsoft Windows"
 	}
 
-	if ($null -eq (Get-Command -Type Application -Name "$SUDOEXE" -ErrorAction Ignore)) {
-		throw "'$SUDOEXE' cannot be found."
+	if (!(Get-Command -Type Application -Name $Env:SUDOEXE -ErrorAction Ignore)) {
+		throw "Env:SUDOEXE is set to '$Env:SUDOEXE' but it cannot be found."
```

Comment:
```suggestion
		throw "Env:SUDOEXE is set to '$Env:SUDOEXE' but cannot be found."
```

Reply:
maybe "that cannot be found." or keep "it"

Reply:
Yeah, I think this:
"Env:SUDOEXE is set to 'C:\Windows\Sudo.exe' but it cannot be found" 

Is better than:
"Env:SUDOEXE is set to 'C:\Windows\Sudo.exe' but cannot be found"